src/core/api/clock.ts

```diff
@@ -345,15 +359,29 @@ export interface IClockOptions {
 function createClock(
   mediaElement : HTMLMediaElement,
   options : IClockOptions
-) : Observable<IClockTick> {
-  return observableDefer(() : Observable<IClockTick> => {
+) : IClockHandler {
+  // Allow us to identify seek performed internally by the player.
+  let isInternalSeekingComing = false;
+  function setCurrentTime(time: number) {
+    isInternalSeekingComing = true;
+    mediaElement.currentTime = time;
+  }
+  const clock$ = observableDefer(() : Observable<IClockTick> => {
     let lastTimings : IClockTick = objectAssign(
       getMediaInfos(mediaElement, "init"),
-      { stalled: null, getCurrentTime: () => mediaElement.currentTime });
-
+      { stalled: null,
+        getCurrentTime: () => mediaElement.currentTime,
+        isInternalSeekingComing: false }
+    );
     function getCurrentClockTick(state : IMediaInfosState) : IClockTick {
-      const mediaTimings = getMediaInfos(mediaElement, state);
+      const tmpState: IMediaInfosState =  state === "seeking" &&
+                                          isInternalSeekingComing ?
+                                            "internal-seeking" : state;                                            
+      const mediaTimings = getMediaInfos(mediaElement, tmpState);
       const stalledState = getStalledStatus(lastTimings, mediaTimings, options);
+      if (state === "seeking" && isInternalSeekingComing) {
```

Comment:
Instead of writing this condition two times, why not just checking it a single time? It would seem more readable to me.

There's no problem at all using `let` somewhere if that's the cleaner way to do it.